### PR TITLE
Add Newton-M2 GPS which is cherry-picked from branch v3.1_dev

### PR DESCRIPTION
**Pull request by lishuanghua**
Add Newton-M2 GPS which is cherry-picked from branch v3.1_dev.

### CODE REVIEW

File: modules/drivers/gnss/parser/BUILD [hunk 1]
```diff
@@ -8,7 +8,8 @@ cc_library(
         ":data_parser",
         ":novatel_parser",
         ":rtcm_parser",
-        ":rtcm3_parser"
+        ":rtcm3_parser",
+        ":newtonm2_parser"
```
Comment (xiaoxq) [inline modules/drivers/gnss/parser/BUILD]: Comma

File: modules/drivers/gnss/parser/data_parser.cc [hunk 2]
```diff
@@ -36,7 +36,7 @@
 #include "modules/drivers/gnss/proto/ins.pb.h"
 #include "modules/drivers/gnss/util/time_conversion.h"
 #include "modules/drivers/gnss/util/utils.h"
-
+#include "modules/drivers/gnss/parser/newtonm2_parser.h"
```
Comment (xiaoxq) [inline modules/drivers/gnss/parser/data_parser.cc]: Keep header sorted

File: modules/drivers/gnss/parser/newtonm2_parser.cc [hunk 3]
```diff
@@ -0,0 +1,916 @@
...
+ * limitations under the License.
+ *****************************************************************************/
+
+// An parser for decoding binary messages from a Starneto Newton-M2 receiver.
+// The following messages must be logged in order for this parser to work
+// properly.
+//
+
+#include "modules/drivers/gnss/parser/newtonm2_parser.h"
+#include <cmath>
```
Comment (xiaoxq) [inline modules/drivers/gnss/parser/newtonm2_parser.cc]: blank line between each section of headers

File: modules/localization/msf/local_integ/measure_republish_process.cc [hunk 4]
```diff
@@ -53,6 +54,78 @@ Status MeasureRepublishProcess::Init(
...
+  if (imu_ant_fin) {
+    bool success = LoadImuGnssAntennaExtrinsic(params.ant_imu_leverarm_file,
+                                               &imu_gnssant_extrinsic_);
+    if (!success) {
+      AERROR << "IntegratedLocalization: Fail to access the lever arm "
+                    "between imu and gnss extrinsic file: "
+                 << params.ant_imu_leverarm_file;
+    }
+    AINFO << "gnss and imu lever arm in vehicle frame: "
+              << " " << imu_gnssant_extrinsic_.ant_num << " "
```
Comment (xiaoxq) [inline modules/localization/msf/local_integ/measure_republish_process.cc]: Align the "<<"

File: modules/localization/msf/local_integ/measure_republish_process.cc [hunk 5]
```diff
@@ -529,6 +622,164 @@ bool MeasureRepublishProcess::CheckBestgnssposeStatus(
...
+  novatel_heading_mutex_.unlock();
+
+  double delta_time_between_height = 0.0;
+  height_mutex_.lock();
+  delta_time_between_height = measure_data->time - map_height_time_;
+  height_mutex_.unlock();
+
+  if (delta_time_between_height < 1.0) {
+    AINFO << "the heading time and delta time: " << std::setprecision(15)
+              << measure_data->time << " " << delta_time_between_height;
```
Comment (mickeyouyou) [inline modules/localization/msf/local_integ/measure_republish_process.cc]: maybe just use one "<<" is enough.

File: modules/localization/msf/local_integ/measure_republish_process.cc [hunk 6]
```diff
@@ -529,6 +622,164 @@ bool MeasureRepublishProcess::CheckBestgnssposeStatus(
...
+    AINFO << "imu_gnssant_extrinsic_: "
+              << imu_gnssant_extrinsic_.transform_2.translation()[0] << ", "
+              << imu_gnssant_extrinsic_.transform_1.translation()[0] << ", "
+              << imu_gnssant_extrinsic_.transform_2.translation()[1] << ", "
+              << imu_gnssant_extrinsic_.transform_2.translation()[1];
+    AINFO << "the yaw between double ant yaw and vehicle: "
+              << imu_ant_yaw_angle;
+  }
+  AINFO << "novatel heading is: " << std::setprecision(15)
+            << measure_data->time << " " << std::setprecision(6) << gnss_yaw;
```
Comment (mickeyouyou) [inline modules/localization/msf/local_integ/measure_republish_process.cc]: <<

File: modules/drivers/gnss/parser/BUILD (hunk 1 above)
Comment (lishuanghua) [reply modules/drivers/gnss/parser/BUILD]: How should I modify it?

File: modules/localization/msf/local_integ/measure_republish_process.cc (hunk 5 above)
Comment (lishuanghua) [reply modules/localization/msf/local_integ/measure_republish_process.cc]: How should I modify it?

File: modules/drivers/gnss/parser/BUILD (hunk 1 above)
Comment (mickeyouyou) [reply modules/drivers/gnss/parser/BUILD]: add comma at the end of line 12.

File: modules/localization/msf/local_integ/measure_republish_process.cc [hunk 7]
```diff
@@ -58,23 +59,23 @@ Status MeasureRepublishProcess::Init(
 
   std::ifstream imu_ant_fin(params.ant_imu_leverarm_file.c_str());
   AINFO << "the ant_imu_leverarm file: "
-            << params.ant_imu_leverarm_file.c_str() << std::endl;
+        << params.ant_imu_leverarm_file.c_str() << std::endl;
```
Comment (mickeyouyou) [inline modules/localization/msf/local_integ/measure_republish_process.cc]: you can remove `std::endl` at line end completely when use apollo log Macro.

Comment (lishuanghua) [reply modules/localization/msf/local_integ/measure_republish_process.cc]: I have Already done it, please review again.

File: modules/drivers/gnss/parser/newtonm2_parser.cc [hunk 8]
```diff
@@ -0,0 +1,916 @@
...
+bool NewtonM2Parser::HandleRawImu(const novatel::RawImu* imu) {
+  double gyro_scale = 0.0;
+  double accel_scale = 0.0;
+  float imu_measurement_span = 1.0 / 200.0;
+
+  if (newtonm2::is_zero(gyro_scale_)) {
+    novatel::ImuParameter param = novatel::GetImuParameter(imu_type_);
+
+    if (newtonm2::is_zero(param.sampling_rate_hz)) {
+      AERROR_EVERY(5) << "Unsupported IMU type ADUS16488.";
```
Comment (mickeyouyou) [inline modules/drivers/gnss/parser/newtonm2_parser.cc]: wrong imu type here. why same error still keep from different PRs? please use ONE PR for your 
 all changes about same topic.

Comment (lishuanghua) [reply modules/drivers/gnss/parser/newtonm2_parser.cc]: I have Already modified it, please review again.

File: modules/localization/msf/local_integ/measure_republish_process.cc (hunk 4 above)
Comment (lishuanghua) [reply modules/localization/msf/local_integ/measure_republish_process.cc]: I have Already modified it, please review again.

File: modules/drivers/gnss/parser/newtonm2_parser.cc (hunk 3 above)
Comment (xiaoxq) [reply modules/drivers/gnss/parser/newtonm2_parser.cc]: Please fix.
